### PR TITLE
Fix stability of SUnitTest's related to the  process monitoring

### PR DESCRIPTION
**Pull request by dionisiydk**
It contributes to #10429

### CODE REVIEW

File: src/SUnit-Tests/SUnitTest.class.st [hunk 1]
```diff
@@ -422,9 +443,11 @@ SUnitTest >> testGreenTestThenLongRunningTest [
 SUnitTest >> testHangedChildProcessTestWhenItIsAllowedToLeaveProcessesAfterTest [
 
 	| case result hangedProcess |
-
+	
 	case := self class selector: #hangedChildProcessTest.	
 	case executionProcessMonitor allowTestToLeaveProcesses.
+	forkedProcesses := OrderedCollection new.
+	case forkedProcesses: forkedProcesses.
```
Comment (guillep) [inline src/SUnit-Tests/SUnitTest.class.st]: Would be nice to not repeat the code between this, `testHangedChildProcessTestWhenLeftProcessIsConsideredAsFailure`,`testHangedTestDueToFailedChildProcess`, `testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAno` and `testFailedChildProcessTest`

**Review by guillep — CHANGES_REQUESTED** [review]

**Comment by MarcusDenker** [comment]
- [CI] Failing randomly: StPlaygroundPageTest>>testContents #10519

**Comment by MarcusDenker** [comment]
```
New failing - 5
windows-64 / Tests-windows-64 / testFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Windows64.SUnit.Tests.Core.SUnitTest
<1s
windows-64 / Tests-windows-64 / testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Windows64.SUnit.Tests.Core.SUnitTest
<1s
windows-64 / Tests-windows-64 / testSmokeTestForDemoPages – Windows64.Spec2.Tests.Examples.SpDemoTest
<1s
windows-64 / Tests-windows-64 / testFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Windows64.SUnit.Tests.Core.SUnitTest
<1s
windows-64 / Tests-windows-64 / testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Windows64.SUnit.Tests.Core.SUnitTest
<1s
Existing failures - 9
unix-64 / Tests-unix-64 / testFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Unix64.SUnit.Tests.Core.SUnitTest
<1s
unix-64 / Tests-unix-64 / testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Unix64.SUnit.Tests.Core.SUnitTest
<1s
unix-64 / Tests-unix-64 / testFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Unix64.SUnit.Tests.Core.SUnitTest
<1s
unix-64 / Tests-unix-64 / testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAnotherTest – Unix64.SUnit.Tests.Core.SUnitTest
<1s
osx-64 / Tests-osx-64 / testDynamicVariable – MacOSX64.Kernel.Tests.Processes.ProcessSpecificTest
10s
osx-64 / Tests-osx-64 / testFailedChildProcessTestShouldNotEffectFollowingAnotherTest – MacOSX64.SUnit.Tests.Core.SUnitTest
<1s
osx-64 / Tests-osx-64 / testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAnotherTest – MacOSX64.SUnit.Tests.Core.SUnitTest
<1s
osx-64 / Tests-osx-64 / testFailedChildProcessTestShouldNotEffectFollowingAnotherTest – MacOSX64.SUnit.Tests.Core.SUnitTest
<1s
osx-64 / Tests-osx-64 / testFailedTestWithFailedChildProcessTestShouldNotEffectFollowingAnotherTest – MacOSX64.SUnit.Tests.Core.SUnitTest
```

**Comment by guillep** [comment]
Thanks!

**Pull request merged**